### DISCUSSION TITLE
multi-position channel on switch

### DISCUSSION THREAD

**Discussion by psicho100pro** [OpenI6X/opentx#231] [Ideas]
Hi,
I prepared other silly questions.
I noticed #229. I have 3pos switches from the beginning. Although it was not possible to set them to 3pos they still worked. So it was a bug?
I ask for another reason. Instead of POT, I use Kevin-Varley Divider, two switches next to each other. They work on only one channel, something like multi-position. There is additional settings in betaflight. So it frees up another channel for something else. (You can have more than 4 switches on the transmitter. Up to 8 on 4 channels. This is just a theoretical level)
Because the firmware monitors the switch channels and the output from them is only LOW, MID and HI it is not possible to achieve the same settings here.
Hardware are all the same analog inputs, both for POT and for switches, 0<3.3V. Am I right?
So is it possible to achieve multi-position on switch channels with any experimental settings? (The divider on the PCB would be removed.) Yes I know in the run of the program then it will not be possible to perceive them as logic states for special functions and so on. In the end, however, we are always interested in output on the betaflight / inav channel.
Regards

**Comment by ajjjjjjjj** [comment]
> Hardware are all the same analog inputs, both for POT and for switches, 0<3.3V. Am I right?

Yes,  all of them are full analogs, although OpenTX divides those two types, so it is a switch: toggle, 2 pos, 3 pos, or pot: with/without detent, multipos. So from code perspective if you want multipos it would be simpler to allow edit pot type. With changing switch to pot it would be a little more complicated.
If you want to experiment on that see: `flysky/board.h` lines from :268,

**Reply by Muhaiminul-Hasan** [reply]
Hi, I upgraded all my switches to 3 position switches, but i can't select "3pos" for any of them except SW_SC,

I tried the following changes and recompiling the firmware using gitpod

flysky/board.h lines from :284
from
#define IS_3POS(x)            ((x) == SW_SC)

to
#define IS_3POS(x)            ((x) == SW_SA || (x) == SW_SB || (x) == SW_SC || (x) == SW_SD)
or,
#define IS_3POS(x)            (1)


doesn't seem to work, can you help ?
or should I change it here ?

radio/src/switches.cpp lines from :131

#if defined(PCBI6X)
  CHECK_2POS(SW_SA);
  CHECK_2POS(SW_SB);
  CHECK_3POS(2, SW_SC);
  CHECK_2POS(SW_SD);
#else
  CHECK_3POS(0, SW_SA);
  CHECK_3POS(1, SW_SB);
  CHECK_3POS(2, SW_SC);
  CHECK_3POS(3, SW_SD);

**Comment by mm3x** [comment]
@psicho100pro , what is exactly scheme you use? Mainly aiming to answer two questions: Do you need to use 2-pole switches? Can you use also 3position switches (on-off-on)?

**Comment by psicho100pro** [comment]
2+2 ; 2+3 ; 3+2 ; 3+3 it is not limited, the connection is the same, in addition one more momentary switch can be added.
Let me point out that I made it up for my purposes, so everyone has to adapt for themselves.
I don't have it graphically processed, I can't even upload any file on this stupid company network, I'll create and upload it in the afternoon.
I can state the values of the resistors. But it doesn't have to be exactly accurate.

| resistors | value |
| --- |----|
| R1 | 16k | 
| R2 | 20k | 
| R3 | 6,8k | 
| R4 | 24k | 
| R5 | 91k | 
| R6 | 47k | 
| R7 | 24k | 
| R8 | 330k | 

image later

**Comment by psicho100pro** [comment]
Assembling a new divider "Kevin-Varley Resistance Divider" and connecting.
![vsc ](https://user-images.githubusercontent.com/97060304/168628215-b1066562-796d-46c6-8b07-fb0e809a1906.png)
We merged the switches in the divider, so we have to split them in betaflight. This is not how it really looks, it's to explain the principle. So the yellow lines are potential positions at the exit of the channel. The blue rectangle characterizes the three positions of the first switch. Red, Pink and Green then the individual positions of the second switch. The last switch and resistors 7 and 8 are reset and it is possible to set, for example, a buzzer to the lowest possible value. But it doesn't have to be connected at all.
![2022-05-16_17h17_05](https://user-images.githubusercontent.com/97060304/168628238-b00cd559-ba36-436f-b432-ec9a697ceb56.png)
Simulations and calculations in the attached file. 
[dělič.xlsx](https://github.com/OpenI6X/opentx/files/8701361/delic.xlsx)
